src/ucp/wireup/select.c

```diff
@@ -1985,13 +2069,16 @@ ucp_wireup_construct_lanes(const ucp_wireup_select_params_t *select_params,
      */
     for (lane = 0; lane < key->num_lanes; ++lane) {
         ucs_assert(select_ctx->lane_descs[lane].lane_types != 0);
+        addr_indices[lane]            = select_ctx->lane_descs[lane].addr_index;
         key->lanes[lane].rsc_index    = select_ctx->lane_descs[lane].rsc_index;
         key->lanes[lane].dst_md_index = select_ctx->lane_descs[lane].dst_md_index;
         key->lanes[lane].dst_sys_dev  = select_ctx->lane_descs[lane].dst_sys_dev;
-        key->lanes[lane].path_index   = select_ctx->lane_descs[lane].path_index;
         key->lanes[lane].lane_types   = select_ctx->lane_descs[lane].lane_types;
         key->lanes[lane].seg_size     = select_ctx->lane_descs[lane].seg_size;
-        addr_indices[lane]            = select_ctx->lane_descs[lane].addr_index;
+        key->lanes[lane].path_index =
+                (select_ctx->lane_descs[lane].path_index ==
+                 UCP_WIREUP_PATH_INDEX_UNDEFINED) ?
+                 0 : select_ctx->lane_descs[lane].path_index;
```

Comment:
maybe helper function for this?
`ucp_wireup_default_path_index(unsigned path_index)`